lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
-}
-
-abstract class RoundedButtonStyle {
-
+class RoundedButtonStyle {
   final Color iconButtonColor;
   final Color backgroundColor;
-
   final BoxShape buttonShape;
   final BorderRadius borderRadius;
-
   final EdgeInsets edgePadding;
   final TextStyle buttonTextStyle;
-
-  final double size;
+  final double height;
+  final double width;
   final double iconEdgePadding;
   final double buttonIconSize;
 
-  RoundedButtonStyle(this.size,
-      this.iconButtonColor, this.backgroundColor, this.iconEdgePadding,
-      {this.buttonShape, this.borderRadius, this.edgePadding,
-        this.buttonTextStyle, this.buttonIconSize});
-}
+  RoundedButtonStyle({this.height, this.width,
+    this.iconButtonColor, this.backgroundColor, this.iconEdgePadding,
+    this.buttonShape, this.borderRadius, this.edgePadding,
+    this.buttonTextStyle, this.buttonIconSize});
+
+  factory RoundedButtonStyle.defaultStyle() {
+    return RoundedButtonStyle(
+        iconButtonColor: Color(0xFFFFFFFF),
+        backgroundColor: Color(0xff24d900),
+        buttonShape: BoxShape.rectangle,
+        borderRadius: BorderRadius.all(Radius.circular(20)),
+        edgePadding: EdgeInsets.all(0),
+        buttonTextStyle: null,
+        height: 24.0,
+        width: 24.0,
+        iconEdgePadding: 0,
+        buttonIconSize: 9.0
+    );
+  }
 
-class _DefaultStyle implements RoundedButtonStyle {
+  factory RoundedButtonStyle.compactRoundedButton() {
+    return RoundedButtonStyle.defaultStyle().copyWith();
+  }
 
-  final Color iconButtonColor =  const Color(0xFFFFFFFF);
-  final Color backgroundColor =  const Color(0xff24d900);
+  factory RoundedButtonStyle.compactBigRoundedButton() {
+    return RoundedButtonStyle.defaultStyle().copyWith(
+        borderRadius: BorderRadius.all(Radius.circular(40)),
+        height: 48.0,
+        width: 48.0,
+        buttonIconSize: 14.0
+    );
+  }
 
-  final BoxShape buttonShape = BoxShape.rectangle;
-  final BorderRadius borderRadius = const BorderRadius.all(Radius.circular(20));
+  factory RoundedButtonStyle.largeRoundedButtonStyle() {
+    return RoundedButtonStyle.defaultStyle().copyWith(
+      borderRadius: BorderRadius.all(Radius.circular(16)),
+      edgePadding: EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32),
+      buttonTextStyle: TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff)),
+      iconEdgePadding: 5,
+      height: 56,
+      width: double.infinity,
+      buttonIconSize: null,
 
-  final EdgeInsets edgePadding = const EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32);
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+    );
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.actionSheetLargeRoundedButton(){
```

Comment:
why does a generic component now know about action sheets? Don't add every style to a generic component! imagine how big this file would get.  It should just have a default style, that's it! 
The other styles should be created at the point of use.... so action sheet should own this.

Reply:
This will be changed in the Jessica PR, because is related with the ActionSheet